tests/test___main__.py

```diff
@@ -6,40 +6,74 @@
 from sqlrunner.__main__ import entrypoint
 
 
-def test_entrypoint(monkeypatch, tmp_path, dsn):
+@pytest.fixture
+def input_file(monkeypatch, tmp_path):
     monkeypatch.chdir(tmp_path)
-    pathlib.Path("input.sql").write_text(
-        f"-- {T1OOS_TABLE} intentionally not excluded\nSELECT 1 AS patient_id",
+
+    path = pathlib.Path("input.sql")
+    path.write_text(
+        f"-- {T1OOS_TABLE} intentionally not excluded\nSELECT 1 AS Patient_ID",
         "utf-8",
     )
-    monkeypatch.setattr(
-        "sys.argv",
-        [
-            "__main__",
-            "--output",
-            "output.csv",
-            "--log-file",
-            "log.json",
-            "input.sql",
-        ],
-    )
+    return str(path)
+
+
+@pytest.mark.parametrize("pass_dummy_data_file", [False, True])
+def test_entrypoint_with_dsn(
+    monkeypatch, tmp_path, dsn, pass_dummy_data_file, input_file
+):
+    monkeypatch.chdir(tmp_path)
     monkeypatch.setattr("os.environ", {"DATABASE_URL": dsn})
+
+    argv = ["__main__", "--output", "output.csv", "--log-file", "log.json"]
+
+    if pass_dummy_data_file:
+        pathlib.Path("dummy_data.csv").touch()  # Notice the dummy data file is empty
+        argv.extend(["--dummy-data-file", "dummy_data.csv"])
+
+    argv.append(input_file)
```

Comment:
Why not put this earlier in `argv`?

Reply:
I found it easier to think of the arguments as

```sh
--output output.csv --log-file log.json --dummy-data-file dummy_data.csv input.sql
```

rather than

```sh
--output output.csv --log-file log.json input.sql --dummy-data-file dummy_data.csv
```